lib/rest-client-view.coffee

```diff
@@ -194,10 +196,13 @@ class RestClientView extends ScrollView
 
   openInEditor: ->
     textResult = $(rest_form.result).text()
-    file_name = "#{current_method} - #{$(rest_form.url).val()}"
-    editor = new RestClientEditor(textResult, file_name)
+    editor = new RestClientEditor(textResult, @constructFileName())
     editor.open()
 
+  constructFileName: ->
+    extension = mime.extension(@lastResponse.headers['content-type']);
+    "#{@lastRequest.method} #{@lastRequest.url}#{'.' + extension if extension}"
```

Comment:
This is kind of unreadable, could you rewrite it in a ways It's easier to read? Than you! :-)